### DISCUSSION TITLE
Azure AD authentication Webmin

### DISCUSSION THREAD

**Discussion by Ilyasbadr** [webmin/webmin#2321] [Q&A] [answered]
hey guys,
I have a simple quick question:
Is it possible to configure webmin to use Azure-AD for authentication??

**Comment by jcameron** [comment]
Is there a PAM module that can check passwords against Azure AD?  If so, maybe it can be done..

**Reply by Ilyasbadr** [reply]
@jcameron Sorry for the late reply unfortunately no PAM module that can check password against Azure AD. I would like to use SAML Authentication if that possible

**Reply by jcameron** [reply]
Sorry, but we don't support SAML in Webmin

**Reply by Ilyasbadr** [reply]
It s a pity, I have one last question regarding ldap authentication is it necessary to join or integrate the Linux machine into the domain?

**Reply by jcameron** [reply]
Not necessarily - it depends on whether you want to just check passwords via LDAP, or have the entire Webmin or Unix user accounts stored in the LDAP server?

**Reply by Ilyasbadr** [reply]
I think is enough just checking passwords via LDAP
The Goal: I want that members of an ad group our domain to be able to log in to Webmin GUI without creating any users in webmin

**Reply by jcameron** [reply]
Do are these users who already exist in AD? If so, what schema are they using?

**Reply by Ilyasbadr** [reply]
yes these Users are already exist in AD and they are Using schema Version 88 = Windows Server 2019 and beyond
by the way I able to browse the ad User in webmin under System  --> LDAP Client --> LDAP Browser
![image](https://github.com/user-attachments/assets/1c1fee8d-9235-458b-b08f-d45334340616)
 but I still can not Use the ad accounts for the authentication :(, any Idea? 
Thank you for your time and Support

**Reply by jcameron** [reply, marked as answer]
So unfortunately the only main way to use users in LDAP to authenticate in Webmin are for them to be Unix users with the standard posixAccount object class, and for your Linux system to be configured via NSS-LDAP to recognize those users. I don't think this is possible if they have whatever the object class is for Windows users..

**Reply by Ilyasbadr** [reply]
thank you very much for your tip after I added posixAccount to object class and I filled in the following attributes, I was able to log in with the ad user

<img width="361" alt="image" src="https://github.com/user-attachments/assets/5ca5a33a-0951-4623-9fd2-c0352a3d41f4">

**Reply by jcameron** [reply]
Excellent, I didn't think it would be that easy!
